crates/server/README.md

```diff
@@ -0,0 +1,257 @@
+
+**7. Get Context**
+ - **Path**: `/contexts/:context_id`
+ - **Method**: `GET`
+ - **Description**: Retrieves details of a specific context by ID.
+
+**8. Get Context Users**
+ - **Path**: `/contexts/:context_id/users`
+ - **Method**: `GET`
+ - **Description**: Lists users associated with a specific context.
+
+**9. Get Context Client Keys**
+ - **Path**: `/contexts/:context_id/client-keys`
+ - **Method**: `GET`
+ - **Description**: Lists client keys for a specific context.
+
+**10. Get Context Storage**
+ - **Path**: `/contexts/:context_id/storage`
+ - **Method**: `GET`
+ - **Description**: Retrieves storage information for a specific context.
+
+**11. List Contexts**
+ - **Path**: `/contexts`
+ - **Method**: `GET`
+ - **Description**: Lists all contexts.
+
+**12. Delete Auth Keys**
+ - **Path**: `/identity/keys`
+ - **Method**: `DELETE`
+ - **Description**: Deletes authentication keys.
+ 
+
+### Unprotected Routes
+These routes do not require authentication.
+
+**1. Health Check**
+ - **Path**: `/health`
+ - **Method**: `GET`
+ - **Description**: Checks the health of the API.
+
+**2. Request Challenge**
+ - **Path**: `/request-challenge`
+ - **Method**: `POST`
+ - **Description**: Requests a challenge for authentication.
+
+**3. Add Client Key**
+ - **Path**: `/add-client-key`
+ - **Method**: `POST`
+ - **Description**: Adds a new client key.
+
+**4. Install Dev Application**
+ - **Path**: `/dev/install-application`
+ - **Method**: `POST`
+ - **Description**: Installs a development application.
+
+**5. Manage Dev Contexts**
+ - **Path**: `/dev/contexts`
+ - Methods: `GET`, `POST`
+ - **Description**: Lists (`GET`) and creates (`POST`) development contexts.
+
+**6. List Dev Applications**
+
+ - **Path**: `/dev/applications`
+ - **Method**: `GET`
+ - **Description**: Lists all development applications.
+
+
+## JSON rpc endpoints
+
+The JSON-RPC server endpoints are structured to handle various request types and are configured based on a JSON-RPC configuration that determines if the server is enabled
+
+**Base path**: `/jsonrpc`
+
+**1. Handle JSON-RPC Request**
+
+- **Path**: `/jsonrpc`
+- **Method**: `POST`
+- **Description**: Handles incoming JSON-RPC requests, which can be `query` or `mutate` requests, processes them, and returns the appropriate response.
+
+
+## Websocket endpoints
+
+The WebSocket, accessible at /ws, allows clients to dynamically subscribe to and unsubscribe from real-time updates about specific contexts within the Calimero Server.
+
+**1. Handle WebSocket Request**
+
+ - **Path**: `/ws`
+ - **Method**: `GET`
+ - **Description**: Handles incoming WebSocket requests, which can be subscribe or unsubscribe requests, processes them, and returns the appropriate response.
+
+
+## Examples
+Examples of Admin Server usage can be found within the [Admin Dashboard](https://github.com/calimero-network/admin-dashboard) and the [Only Peers example](https://github.com/calimero-network/only-peers-client
```

Comment:
All communication with the node should be exposed through `calimero sdk`.

Reply:
Updated